### PR TITLE
Delete confirmation

### PR DESCRIPTION
**Pull request by lkudyba**
@quirk0o @jtkpiotr @mhib @adamzima @pbzymek @s1mplex

### CODE REVIEW

File: client/app/components/calendar/event/DeleteButton.jsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import React, { PropTypes } from 'react';
-import { Tooltip, Overlay } from 'react-bootstrap';
-import { If } from 'react-if';
+import { Tooltip, Overlay, Button, Modal } from 'react-bootstrap';
+import { If, Else } from 'react-if';
+import bindAll from 'lodash/bindAll';
```
Comment (quirk0o) [inline client/app/components/calendar/event/DeleteButton.jsx]: 👍

File: client/app/components/calendar/event/DeleteButton.jsx [hunk 2]
```diff
@@ -59,4 +84,11 @@ export default class DeleteButton extends React.Component {
...
     return false;
   }
+
+  _handleOnClick() {
+    if (!this.props.disabled) {
+      return this._handleOnClickEnabled();
+    }
+    return this._handleOnClickDisabled();
+  }
 }
```
Comment (quirk0o) [inline client/app/components/calendar/event/DeleteButton.jsx:19]: Puke
You're putting all the logic and everything in one component. `DeleteButton` should be a simple presentational component. And now it contains a modal and God knows what else ^^

Comment (mhib) [inline client/app/components/calendar/event/DeleteButton.jsx:19]: Don't use toxic words.

File: client/app/components/calendar/event/EventDestroyer.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+import { If, Else } from 'react-if';
+import bindAll from 'lodash/bindAll';
+import DeleteButton from 'components/calendar/event/DeleteButton';
+import DeleteTooltip from 'components/calendar/event/DeleteTooltip';
+import DeleteConfirmation from 'components/calendar/event/DeleteConfirmation';
+import './event.scss';
+
+
```
Comment (adamzima) [inline client/app/components/calendar/event/EventDestroyer.jsx]: empty line #importantComment

File: client/app/components/calendar/event/EventDestroyer.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+export default class EventDestroyer extends React.Component {
```
Comment (adamzima) [inline client/app/components/calendar/event/EventDestroyer.jsx:9]: <3

File: client/app/components/calendar/event/EventDestroyer.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    );
+  }
+
+  _hideConfirmationModal() {
+    this.setState({ showConfirmationModal: false });
+  }
+
+  _handleConfirmDeletion() {
+    this._hideConfirmationModal();
+    return this.props.onDelete();
```
Comment (adamzima) [inline client/app/components/calendar/event/EventDestroyer.jsx:49]: Do we need return here?

File: client/app/components/calendar/event/EventDestroyer.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  _handleOnClickEnabled() {
+    this.setState({ showConfirmationModal: true });
+  }
+
+  _handleOnClickDisabled() {
+    this.setState({
+      showIndicator: true
```
Comment (adamzima) [inline client/app/components/calendar/event/EventDestroyer.jsx]: I'm just curious - why here you split the line and in `_handleOnClickEnabled` you didn't.

File: client/app/components/calendar/event/EventDestroyer.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    setTimeout(() => this.setState({ showIndicator: false }), 2000);
+
+    return false;
+  }
+
+  _handleOnClick() {
+    if (!this.props.disabled) {
+      return this._handleOnClickEnabled();
+    }
+    return this._handleOnClickDisabled();
```
Comment (adamzima) [inline client/app/components/calendar/event/EventDestroyer.jsx:68]: Again, I think you don't need to return anything from this function.

**Pull request merged**